reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,345 @@
+sed -i -e "s#^OPTIONS='--selinux-enabled'#OPTIONS='--selinux-enabled --insecure-registry 172.30.0.0/16'#" /etc/sysconfig/docker
+                                                                                         
+cat <<EOF > /etc/sysconfig/docker-storage-setup
+DEVS=/dev/sdc
+VG=docker-vg
+EOF
+
+docker-storage-setup                                                                                                                                    
+systemctl enable docker
+systemctl start docker
+
+
+cat <<EOF > /etc/ansible/hosts
+[OSEv3:children]
+masters
+etcd
+nodes
+misc
+
+[OSEv3:vars]
+azure_resource_group=${RESOURCEGROUP}
+rhn_pool_id=${RHNPOOLID}
+openshift_install_examples=true
+deployment_type=openshift-enterprise
+openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
+
+# default selectors for router and registry services
+openshift_router_selector='region=infra'
+openshift_registry_selector='region=infra'
+
+ansible_become=yes
+ansible_ssh_user=${AUSERNAME}
+remote_user=${AUSERNAME}
+
+openshift_master_default_subdomain=${ROUTEREXTIP}.xip.io 
+openshift_use_dnsmasq=False
+openshift_public_hostname=${RESOURCEGROUP}.trafficmanager.net
+
+openshift_master_cluster_method=native
+openshift_master_cluster_hostname=${RESOURCEGROUP}.trafficmanager.net
+openshift_master_cluster_public_hostname=${RESOURCEGROUP}.trafficmanager.net
+
+# Enable cockpit
+osm_use_cockpit=true
+
+# Set cockpit plugins
+osm_cockpit_plugins=['cockpit-kubernetes']
+
+# default storage plugin dependencies to install, by default the ceph and
+# glusterfs plugin dependencies will be installed, if available.
+osn_storage_plugin_deps=['iscsi']
+
+[masters]
+master1 openshift_node_labels="{'role': 'master'}"
+master2 openshift_node_labels="{'role': 'master'}"
+master3 openshift_node_labels="{'role': 'master'}"
+
+[etcd]
+master1
+master2
+master3
+
+[nodes]
+master1 openshift_node_labels="{'region':'master','zone':'default'}" 
+master2 openshift_node_labels="{'region':'master','zone':'default'}" 
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+infranode1 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode2 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode3 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+
+[quotanodes]
+master1 openshift_node_labels="{'region':'master','zone':'default'}" 
+master2 openshift_node_labels="{'region':'master','zone':'default'}" 
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+
+[misc]
+store1
+EOF
+
+
+cat <<EOF > /home/${AUSERNAME}/subscribe.yml
+---
+- hosts: all
+  vars:
+    description: "Subscribe OSE"
+  tasks:
```

Comment:
This is all tested multiple times, with retry logic etc that has proven to work every time.
I've tried a variety of ways of doing this, and current logic resolves all. This includes rhn and ansible bugs, and hangs. Single Threaded stil ldoes not "solve" the varous and wonderful rhn and subscription manager issues, the retry logic works well.